src/main/java/jeff/task/TaskList.java

```diff
@@ -0,0 +1,195 @@
+                        if (taskParts.length >= 5) {
+                            currentTask = new EventTask(taskDescription,
+                                    LocalDateTime.parse(taskParts[3], DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm")),
+                                    LocalDateTime.parse(taskParts[4], DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm")),
+                                    isDone);
+                        }
+                        break;
+
+                    default:
+                        break;
+                    }
+                } catch (DateTimeParseException e) {
+                    throw new FileCorruptException();
+                }
+
+                // Add task to task list if it exists
+                if (currentTask != null) {
+                    this.tasks.add(currentTask);
+                } else {
+                    throw new FileCorruptException();
+                }
+            }
+
+        } catch (FileCorruptException e) {
+            throw new JeffException(e.toString());
+        }
+    }
+
+    /**
+     * Returns the task at the given index in the task list.
+     *
+     * @param index Index of the target task.
+     * @return Task at the given index.
+     */
+    public Task get(int index) {
+        return this.tasks.get(index);
+    }
+
+    /**
+     * Adds the given task to the task list.
+     *
+     * @param task Task to be added to the task list.
+     */
+    public void add(Task task) {
+        this.tasks.add(task);
+    }
+
+    /**
+     * Removes the given task from the task list.
+     *
+     * @param task Task to be removed from the task list.
+     */
+    public void remove(Task task) {
+        this.tasks.remove(task);
+    }
+
+    /**
+     * Returns the size of the task list.
+     *
+     * @return Size of the task list.
+     */
+    public int size() {
+        return this.tasks.size();
+    }
+
+    /**
+     * Checks if the task list is empty or not.
+     *
+     * @return true if the task list is empty and false otherwise.
+     */
+    public boolean isEmpty() {
+        return this.tasks.isEmpty();
+    }
+
+    /**
+     * Returns a list of tasks that only takes place on the given date.
+     *
+     * @param date Given date.
+     * @return A list of filtered tasks.
+     */
+    public List<Task> filterByDate(LocalDate date) {
+        return this.tasks.stream().filter(task -> task.isOnThisDate(date)).toList();
+    }
+
+    /**
+     * Returns a list of tasks filtered by the given name.
+     *
+     * @param name Given name to filter.
+     * @return List of filtered tasks.
+     */
+    public List<Task> filterByName(String name) {
+        return this.tasks.stream().filter(task -> task.contains(name)).toList();
+    }
+
+    /**
+     * Returns a list of the file string representation of the tasks.
+     *
+     * @return A list of the file string representation of the tasks.
+     */
+    public List<String> toFileStrings() {
```

Comment:
Is there a clearer name for `toFileStrings`, since it is returning not a String but a List.

Reply:
Thanks for pointing it out! I will try to change it to make it clearer.